src/affine.rs

```diff
@@ -0,0 +1,212 @@
+use nalgebra::{Matrix3, Matrix4, Quaternion, RowVector4, Scalar, SymmetricEigen, Vector3, U1, U3};
+
+pub type Affine3 = Matrix3<f32>;
+pub type Affine4 = Matrix4<f32>;
+
+const QUARTERNION_THRESHOLD: f32 = -::std::f32::EPSILON * 3.0;
+
+/// Separate a 4x4 affine into its 3x3 affine and translation components.
+pub fn get_affine_and_translation<T: Scalar>(affine: &Matrix4<T>) -> (Matrix3<T>, Vector3<T>) {
```

Comment:
Non-idiomatic `get_` prefix. This operation seems to _convert_ something more than it _gets_ something, and even as a getter, it should not start with `get_` as per the [C-GETTER](https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-getter) guidelines. ¹